src/controllers/engine/controllerengine.cpp

```diff
@@ -319,14 +338,17 @@ void ControllerEngine::initializeScripts(const QList<ControllerPreset::ScriptFil
     args << QJSValue(m_pController->getName());
     args << QJSValue(ControllerDebug::enabled());
 
-    // Call the init method for all the prefixes.
-    bool success = callFunctionOnObjects(m_scriptFunctionPrefixes, "init", args, true);
+    // for legacy mappings
+    callFunctionOnObjects(m_scriptFunctionPrefixes, "init", args, true);
+}
 
-    // We failed to initialize the controller scripts, shutdown the script
-    // engine to avoid error popups on every button press or slider move
-    if (!success) {
-        gracefulShutdown();
-        uninitializeScriptEngine();
+void ControllerEngine::receiveInput(const QByteArray& data, mixxx::Duration timestamp) {
+    // Legacy mappings do not use this, so do not VERIFY_OR_DEBUG_ASSERT here.
+    if (m_inputCallback.isCallable()) {
+        QJSValueList args;
+        args << byteArrayToScriptValue(data);
+        args << static_cast<uint>(timestamp.toIntegerMillis());
```

Comment:
Is this an appropriate level of precision? Is any more precision actually accurate? @Pegasus-RPG